geoengine/raster.py

```diff
@@ -142,16 +146,29 @@ def coords_y(self, pixel_center=False) -> np.ndarray:
     def to_xarray(self, clip_with_bounds: Optional[gety.SpatialBounds] = None) -> xr.DataArray:
         '''
         Return the raster tile as an xarray.DataArray.
-        Xarray does not support masked arrays.
-        Masked pixels are converted to NaNs and the nodata value is set to NaN as well.
+
+        Note:
+            - Xarray does not support masked arrays.
+                - Masked pixels are converted to NaNs and the nodata value is set to NaN as well.
+            - Xarray uses numpy's datetime64[ns] which only covers the years from 1678 to 2262.
+                - Date times that are outside of the defined range are clipped to the limits of the range.
         '''
+
+        # Define the min and max dates. There is no straightforward way to compute this, so we hardocde it.
+        min_date = np.datetime64('1678-09-21 00:12:43.145224192', 'ns')
+        max_date = np.datetime64('2262-04-11 23:47:16.854775807', 'ns')
+
+        # Clip the dates to the min and max range
+        clipped_date = np.clip(self.time_start_ms, min_date, max_date)
```

Comment:
I'm not sure this is correct:

```Python
>>> min_date = np.datetime64('1678-09-21 00:12:43.145224192', 'ns')
>>> max_date = np.datetime64('2262-04-11 23:47:16.854775807', 'ns')

>>> min_date
numpy.datetime64('1678-09-21T00:12:43.145224192')
>>> max_date
numpy.datetime64('2262-04-11T23:47:16.854775807')

>>> my_date = np.datetime64('1500-09-21', 'ms')
>>> my_date
numpy.datetime64('1500-09-21T00:00:00.000')

>>> np.clip(my_date, min_date, max_date)
numpy.datetime64('2085-04-09T23:34:33.709551616')

>>> np.datetime64('1500-09-21T00:00:00.000', 'ns')
numpy.datetime64('2085-04-09T23:34:33.709551616')
```

Reply:
a26357d61e49e44a9dc4bef6c0aa244363343ce6